rcl/include/rcl/graph.h

```diff
@@ -440,6 +441,39 @@ rcl_get_node_names(
   rcutils_string_array_t * node_names,
   rcutils_string_array_t * node_namespaces);
 
+/// Return a list of available nodes in the ROS graph, including their security context names.
+/**
+ * \sa Works like rcl_get_node_names, but it also include the security context name the node is
+ * using as an output.
+ *
+ * <hr>
+ * Attribute          | Adherence
+ * ------------------ | -------------
+ * Allocates Memory   | Yes
+ * Thread-Safe        | No
+ * Uses Atomics       | No
+ * Lock-Free          | Maybe [1]
+ * <i>[1] implementation may need to protect the data structure with a lock</i>
```

Comment:
@ivanpauno what implementations? This may need some qualification.

Reply:
Preexisting, see https://github.com/ros2/rcl/blob/f242967bf024804b3ed1e8b00ebf3b3b3c9541cd/rcl/include/rcl/graph.h#L456 and documentation of the other functions too.

IMO, the comment is clear.
It means that it might lock or not depending of the rmw implementation, and that the user shouldn't make any assumption.

Reply:
I'd be explicit and `RMW`, but fine with it for now.

Reply:
AFAIK, every of them are locking now (at least, the three default implementations).
It's pretty hard to have a complete list, considering that there are also external rmw implementations.

Reply:
Oh, for sure, I wouldn't mention every existing RMW implementation, I'd just qualify the noun as in:
```suggestion
 * <i>[1] RMW implementation in use may need to protect the data structure with a lock</i>
```

Reply:
Done in https://github.com/ros2/rcl/pull/515/commits/35ec36eb170bb5801c17f03637e2a9b97f13ded7.